examples/hotrod/pkg/tracing/rpcmetrics/observer_test.go

```diff
@@ -106,23 +104,18 @@ func TestObserver(t *testing.T) {
 
 func TestTags(t *testing.T) {
 	type tagTestCase struct {
-		key     string
 		variant string
-		value   interface{}
+		attr    attribute.KeyValue
 		metrics []u.ExpectedMetric
 	}
 
 	testCases := []tagTestCase{
-		{key: "something", value: 42, metrics: []u.ExpectedMetric{
+		{attr: attribute.Key("something").Int(42), metrics: []u.ExpectedMetric{
 			{Name: "requests", Value: 1, Tags: tags("error", "false")},
 		}},
-		{key: "error", value: true, metrics: []u.ExpectedMetric{
+		{attr: attribute.Key("error").Bool(true), metrics: []u.ExpectedMetric{
 			{Name: "requests", Value: 1, Tags: tags("error", "true")},
 		}},
```

Comment:
this is not going to work anymore. The new code in the observer generates error metrics based on span status
```
	if status := sp.Status(); status.Code == codes.Error {
		mets.RequestCountFailures.Inc(1)
		mets.RequestLatencyFailures.Record(latency)
```
not based on error=true tag that is no longer used/recognized in OTEL instrumentation

Reply:
so you need to generate spans with/without error status in order to test different metrics. I am not sure if something=42 has any bearing anymore, it's probably just a placehoder. But you do need the `attr` in order to test `"http.status_code"` tag.